### PR TITLE
implement starknet-rs support for block endpoints

### LINKED ISSUE #14: Implement `starknet.rs` support for block endpoints
Implement autogenerated `starknet.rs` support for the following endpoints.

- `starknet_getBlockWithTxHashes`
- `starknet_getBlockWithTxs`
- `starknet_blockNumber`
- `starknet_blockHashAndNumber`
- `starknet_getStateUpdate`
- `starknet_getStorageAt`

**Additional Context:**

- https://github.com/NethermindEth/rpc-request-builder/issues/2
- https://github.com/NethermindEth/rpc-request-builder/blob/main/src/api/rpcspec/methods.ts

### PR DESCRIPTION
**Pull request by Prabhat1308**
This PR implements the ```starknet-rs``` support for block endpoints. 
Fixes #14

### CODE REVIEW

File: src/api/rpcspec/methods.ts [hunk 1]
```diff
@@ -210,7 +212,36 @@ const ReadMethods = [
 });
     `,
     starknetGo: ``,
-    starknetRs: ``,
+    starknetRs: `
+    use starknet::{
+      core::types::{BlockId, BlockTag,MaybePendingBlockWithTxHashes},
```
Comment (jelilat) [inline src/api/rpcspec/methods.ts:217]:
```suggestion
      core::types::{BlockId, BlockTag},
```

File: src/api/rpcspec/methods.ts [hunk 2]
```diff
@@ -224,7 +255,34 @@ const ReadMethods = [
 });
     `,
     starknetGo: ``,
-    starknetRs: ``,
+    starknetRs: `
+    use starknet::{
+      core::types::{BlockId, BlockTag,MaybePendingBlockWithTxs},
```
Comment (jelilat) [inline src/api/rpcspec/methods.ts:260]:
```suggestion
      core::types::{BlockId, BlockTag},
```

File: src/api/rpcspec/methods.ts [hunk 3]
```diff
@@ -464,10 +594,36 @@ const ReadMethods = [
...
     starknetGo: ``,
-    starknetRs: ``,
+    starknetRs: `
+    use starknet::{
+    core::types::{BlockHashAndNumber}
+    providers::{
+      jsonrpc::{HttpTransport, JsonRpcClient},
+      Provider, Url,
+      },
+    };
```
Comment (jelilat) [inline src/api/rpcspec/methods.ts:607]:
```suggestion
    starknetRs: `use starknet::providers::{
    jsonrpc::{HttpTransport, JsonRpcClient},
    Provider, Url,
};
```

File: src/api/rpcspec/methods.ts [hunk 4]
```diff
@@ -258,7 +340,32 @@ const ReadMethods = [
...
     starknetGo: ``,
-    starknetRs: ``,
+    starknetRs: `
+    use starknet::{
+      core::types::FieldElement,
+      providers::{
+      jsonrpc::{HttpTransport, JsonRpcClient},
+      Provider, Url,
+  },
+  };
```
Comment (jelilat) [inline src/api/rpcspec/methods.ts:350]:
```suggestion
    starknetRs: `use starknet::{
    core::types::{BlockId, BlockTag},
    macros::felt,
    providers::{
        jsonrpc::{HttpTransport, JsonRpcClient},
        Provider, Url,
    },
};
```

File: src/components/constant.ts [hunk 5]
```diff
@@ -38,7 +38,30 @@ func main() {
 
 	fmt.Println("SpecVersion:", specVersion)
 }`;
-export const DEFAULT_STARKNET_RS_REQUEST = ``;
+export const DEFAULT_STARKNET_RS_REQUEST = `use starknet::{
+	providers::{
+	  jsonrpc::{HttpTransport, JsonRpcClient},
+	  Provider, Url,
+	},
+  };
```
Comment (jelilat) [inline src/components/constant.ts:46]:
```suggestion
  };
  
```

File: src/api/rpcspec/methods.ts [hunk 6]
```diff
@@ -455,7 +562,30 @@ const ReadMethods = [
 });
     `,
     starknetGo: ``,
-    starknetRs: ``,
+    starknetRs: `
+    use starknet::providers::{
```
Comment (jelilat) [inline src/api/rpcspec/methods.ts:566]:
```suggestion
    starknetRs: `use starknet::providers::{
```

File: src/api/rpcspec/methods.ts [hunk 7]
```diff
@@ -210,7 +212,36 @@ const ReadMethods = [
 });
     `,
     starknetGo: ``,
-    starknetRs: ``,
+    starknetRs: `
+    use starknet::{
```
Comment (jelilat) [inline src/api/rpcspec/methods.ts:216]:
```suggestion
    starknetRs: `use starknet::{
```

File: src/api/rpcspec/methods.ts [hunk 8]
```diff
@@ -224,7 +255,34 @@ const ReadMethods = [
...
```
Comment (jelilat) [inline src/api/rpcspec/methods.ts:259]:
```suggestion
    starknetRs: `use starknet::{
```

File: src/api/rpcspec/methods.ts [hunk 9]
```diff
@@ -238,7 +296,31 @@ const ReadMethods = [
 });
     `,
     starknetGo: ``,
-    starknetRs: ``,
+    starknetRs: `
+      use starknet::{
```
Comment (jelilat) [inline src/api/rpcspec/methods.ts:300]:
```suggestion
    starknetRs: `use starknet::{
```

**Review by jelilat — CHANGES_REQUESTED** [review]
Thank you for your contribution. I made some minor changes recommendation

**Comment by Prabhat1308** [comment]
@jelilat I have made the changes please review.

**Review by jelilat — APPROVED** [review]

**Comment by jelilat** [comment]
Thank you for your contribution @Prabhat1308

**Pull request merged**